src/Test/StateMachine/Internal/Types.hs

```diff
@@ -79,6 +80,12 @@ deriving instance Read (Untyped act) => Read (ParallelProgram act)
 data Fork a = Fork a a a
   deriving (Eq, Functor, Show, Ord, Read)
 
+data ParallelProgram' act = ParallelProgram' (Program act) [Program act]
```

Comment:
Is this going to replace the `Fork`-based `ParallelProgram`?

Reply:
Yeah, that's the idea.